### ISSUE TITLE
Immediate Auto-Analysis does not work with scoped logging

### ISSUE THREAD

**Issue by dstewart-kb4** [reportportal/reportportal#2440]
**Describe the bug**
I have tried to enable Immediate Auto-Analysis using the documentation from https://reportportal.io/docs/analysis/ImmediateAutoAnalysis/. I have been able to successfully enable this using our C#/Selenium based framework, however, when using Python/Behave which uses a BDD style test with scoped logging, the immediate auto-analysis is not working. I've added teh `immediateAutoAnalysis` attribute on the request that occurs for the beginning of a `Scenario` which is the `step` level API call. I've noticed in the logs for the API service the following lines
```
2024-12-12 20:38:35.282 INFO 1 --- [istener-exec973] c.e.t.r.c.a.a.indexer.BatchLogIndexer : Indexing started for 1 items.
2024-12-12 20:38:35.299 INFO 1 --- [istener-exec973] c.e.t.r.c.a.a.indexer.BatchLogIndexer : Indexing of 0 logs is finished for 1 items. 
2024-12-12 20:38:35.307 INFO 1 --- [istener-exec973] c.e.t.r.c.a.a.impl.AnalyzerServiceImpl : Start analysis of '1' items for launch with id '885751'
2024-12-12 20:38:35.316 INFO 1 --- [istener-exec973] c.e.t.r.c.a.a.indexer.BatchLogIndexer : Indexing started for 1 items.
2024-12-12 20:38:35.321 INFO 1 --- [istener-exec973] c.e.t.r.c.a.a.indexer.BatchLogIndexer : Indexing of 0 logs is finished for 1 items.
```
That final log line there leads me to believe that the auto analyzer service is not picking up the scoped logs since when the analyzer performs it's job correctly there with be a number >0 in that line.
One final thing I've noticed is that with immediate auto-analyzer enabled, the auto-analyzer is not running at the end of the test run so there are no failures being marked as their proper issues, whereas, if I remove the line to add the immediate auto analyzer attribute, I then will see the proper issues being marked on test suite completion.

**Expected behavior**
Failures are analyzed as they occur during the test run

**Actual behavior**
Failures are not being properly analyzed during test run time and Auto-Analysis is not kicked off again at test completion, meaning failed tests are not being marked as issues at all.

**Versions:**
 - Version of RP: 24.2 (5.12.0, API: 5.12.1)
 - Behave: 1.2.6
 - behave-reportportal: 4.0.2
 - reportportal-client: 5.5.6

**Comment by HardNorth** [comment]
@dstewart-kb4 What do you mean saying `scoped logging`? I can't recall such feature.

**Comment by dstewart-kb4** [comment]
@HardNorth Sorry 'scoped logging' is what our team refers to it internally, I am talking about nested steps

**Comment by dstewart-kb4** [comment]
On further investigation, this is probably not related to nested steps as I was able to get immediate auto-analysis working in a C# project using Reqnroll, however, I am still seeing the same issue with Python and Behave where in the API service logs it is giving the log `Indexing of 0 logs is finished for 1 items.` and then I am not seeing anything happening in the analyzer service

**Comment by dstewart-kb4** [comment]
I've been able to get the immediate auto analysis to work occasionally, but I still see for some of the results that message about 0 logs. However, once the test suite is finished, if I manually kick off an auto analysis then it will properly mark those failed test results.

**Comment by dstewart-kb4** [comment]
One other thing I've noticed, if a test has been retried the immediate auto analysis does not run on the retries. I can see it running on the first execution of a test, but any later retries of the failure it is not running auto analysis on them.

**Comment by dstewart-kb4** [comment]
@HardNorth Have you had a chance to look into this issue?

**Comment by dstewart-kb4** [comment]
I am wondering if this is a result of async logging, the error log isn't necessarily available at the time of test item completion and therefore the analyzer is not picking up any logs to analyze

**Comment by HardNorth** [comment]
@dstewart-kb4, No, this is rather complicate to reproduce and I'm currently busy with other tasks.
